src/VisualStudio/Core/Def/RoslynDocumentProvider.cs

```diff
@@ -23,9 +26,50 @@ internal class RoslynDocumentProvider : DocumentProvider
 
         protected override void OnBeforeDocumentWindowShow(IVsWindowFrame frame, DocumentId id, bool firstShow)
         {
-            if (_documentTrackingService != null)
+            base.OnBeforeDocumentWindowShow(frame, id, firstShow);
+
+            _documentTrackingService?.DocumentFrameShowing(frame, id, firstShow);
+        }
+
+        protected override void OnBeforeNonRoslynDocumentWindowShow(IVsWindowFrame frame, bool firstShow)
+        {
+            base.OnBeforeNonRoslynDocumentWindowShow(frame, firstShow);
+
+            if (!firstShow)
+            {
+                return;
+            }
+
+            var view = GetTextViewFromFrame(frame);
+            if (view != null)
+            {
+                _documentTrackingService?.OnNonRoslynViewOpened(view);
+            }
+        }
+
+        private ITextView GetTextViewFromFrame(IVsWindowFrame frame)
+        {
+            IntPtr unknown;
+            if (ErrorHandler.Failed(frame.QueryViewInterface(typeof(IVsCodeWindow).GUID, out unknown)) || unknown == IntPtr.Zero)
```

Comment:
This isn't the right way to get a code window from a frame. Microsoft.VisualStudio.Shell.VsShellUtilities has a GetTextView implementation which you should study. It deals with things such as split XAML/Designer views.